### PR TITLE
[WIP] Dependencies update

### PR DESCRIPTION
**Pull request by ghost**
I tested build with
```
sudo apt update && sudo apt install git build-essential cmake pkg-config libtool gperf
git clone https://github.com/yorha-0x/aeon.git
git checkout --track origin/dependencies-update
git submodule update --init
make depends target=x86_64-pc-linux-gnu
```
on clean Ubuntu 20.04 (LTS) x64 it works. Other OS would be good to test.

### CODE REVIEW

**Comment by ghost** [comment]
Would like to have more review on this with other operating systems.

**Comment by stoffu** [comment]
Please modify the commit messages so that they all have the /monero#NNNN at the end. This is helpful later for checking which PRs are already in or not.

File: contrib/depends/packages/unwind.mk [hunk 1]
```diff
@@ -1,12 +1,10 @@
 package=unwind
-$(package)_version=1.2
+$(package)_version=1.5.0
```
Comment (stoffu) [inline contrib/depends/packages/unwind.mk:2]: Is this patch your original? What does "/libunwind#7249" refer to? Libunwind doesn't have that PR number, and Monero's PR 7249 is on something completely different.

If this patch is your original and is also applicable to Monero, please send a PR to Monero first and refer to it here.
https://github.com/aeonix/aeon/blob/master/CONTRIBUTING.md#proposing-changes

File: contrib/depends/packages/expat.mk [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package=expat
 $(package)_version=2.2.4
-$(package)_download_path=https://downloads.sourceforge.net/project/expat/expat/$($(package)_version)
+$(package)_download_path=https://github.com/libexpat/libexpat/releases/download/R_2_2_4
```
Comment (stoffu) [inline contrib/depends/packages/expat.mk:3]: Please add /monero#6500 at the end of the commit message.

File: contrib/depends/packages/qt.mk [hunk 3]
```diff
@@ -1,18 +1,18 @@
 PACKAGE=qt
-$(package)_version=5.7.1
```
Comment (stoffu) [inline contrib/depends/packages/qt.mk:2]: Please add /monero#6822 to the end of the commit message.

File: CMakeLists.txt [hunk 4]
```diff
@@ -440,8 +440,8 @@ if (CMAKE_SYSTEM_NAME MATCHES "(SunOS|Solaris)")
 endif ()
 
 if (APPLE AND NOT IOS)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=x86-64 -fvisibility=default")
-  if (NOT OpenSSL_DIR)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=x86-64 -fvisibility=default -std=c++11")
```
Comment (stoffu) [inline CMakeLists.txt:443]: Hmm, this change is not included in Monero PR 6691 but in 5930. I feel a bit uneasy when someone sneaks in some unrelated change in a cherry-pick, and I like it better if 5930 is cherry-picked beforehand (if it's easy). But okay if you insist.

File: contrib/depends/packages/openssl.mk [hunk 5]
```diff
@@ -1,8 +1,8 @@
 package=openssl
-$(package)_version=1.0.2r
```
Comment (stoffu) [inline contrib/depends/packages/openssl.mk]: This patch looks quite different from the original https://github.com/LMDB/bitmonero/commit/5b30203fe956990e9cf0fa7ad5a280b88b7683e8, and notably misses some configurations related to ARM building. This is why I prefer avoiding out-of-order cherry-picking.

Do you insist that this OpenSSL update is necessary now? If so, maybe it's better not to cherry-pick and make a minimal patch bumping the version only, so that the upstream changes can be picked up later.

File: contrib/depends/funcs.mk [hunk 6]
```diff
@@ -129,9 +129,9 @@ $(1)_config_env+=$($(1)_config_env_$(host_arch)_$(host_os)) $($(1)_config_env_$(
 
 $(1)_config_env+=PKG_CONFIG_LIBDIR=$($($(1)_type)_prefix)/lib/pkgconfig
 $(1)_config_env+=PKG_CONFIG_PATH=$($($(1)_type)_prefix)/share/pkgconfig
-$(1)_config_env+=PATH=$(build_prefix)/bin:$(PATH)
-$(1)_build_env+=PATH=$(build_prefix)/bin:$(PATH)
-$(1)_stage_env+=PATH=$(build_prefix)/bin:$(PATH)
+$(1)_config_env+=PATH="$(build_prefix)/bin:$(PATH)"
+$(1)_build_env+=PATH="$(build_prefix)/bin:$(PATH)"
+$(1)_stage_env+=PATH="$(build_prefix)/bin:$(PATH)"
```
Comment (stoffu) [inline contrib/depends/funcs.mk:134]: Please add /monero#6255 to the end of the commit message.

File: contrib/depends/packages/hidapi.mk [hunk 7]
```diff
@@ -1,8 +1,8 @@
 package=hidapi
-$(package)_version=0.8.0-rc1
-$(package)_download_path=https://github.com/signal11/hidapi/archive
+$(package)_version=0.9.0
+$(package)_download_path=https://github.com/libusb/hidapi/archive
 $(package)_file_name=$(package)-$($(package)_version).tar.gz
-$(package)_sha256_hash=3c147200bf48a04c1e927cd81589c5ddceff61e6dac137a605f6ac9793f4af61
+$(package)_sha256_hash=630ee1834bdd5c5761ab079fd04f463a89585df8fcae51a7bfe4229b1e02a652
```
Comment (stoffu) [inline contrib/depends/packages/hidapi.mk:5]: Please add /monero#5704 to the end of the first line of the commit message.

File: contrib/depends/packages/ldns.mk [hunk 8]
```diff
@@ -1,8 +1,8 @@
 package=ldns
-$(package)_version=1.6.17
+$(package)_version=1.7.1
```
Comment (stoffu) [inline contrib/depends/packages/ldns.mk:2]: This patch also looks different from the original. If it's your original, please make a PR on Monero first.

If your cherry-picking results in a lot of merge conflicts and you have to modify the original patch significantly, then consider redoing it in the original order.

File: contrib/depends/packages/unwind.mk [hunk 9]
```diff
@@ -3,6 +3,11 @@ $(package)_version=1.2
 $(package)_download_path=http://download.savannah.nongnu.org/releases/libunwind
 $(package)_file_name=lib$(package)-$($(package)_version).tar.gz
 $(package)_sha256_hash=1de38ffbdc88bd694d10081865871cd2bfbb02ad8ef9e1606aee18d65532b992
+$(package)_patches=fix_obj_order.patch
+
+define $(package)_preprocess_cmds
+  patch -p0 < $($(package)_patch_dir)/fix_obj_order.patch
+endef
```
Comment (stoffu) [inline contrib/depends/packages/unwind.mk:8]: Please add /monero#6118 to the end of the commit message.

File: contrib/depends/packages/openssl.mk [hunk 10]
```diff
@@ -1,8 +1,8 @@
...
-$(package)_download_path=https://ftp.openssl.org/source/old/1.0.2
+$(package)_version=1.1.1i
+$(package)_download_path=https://www.openssl.org/source
```
Comment (stoffu) [inline contrib/depends/packages/openssl.mk:3]: This patch looks quite different from the original https://github.com/monero-project/monero/pull/7100/commits/5b30203fe956990e9cf0fa7ad5a280b88b7683e8, and notably misses some configurations related to ARM building. This is why I prefer avoiding out-of-order cherry-picking.

Do you insist that this OpenSSL update is necessary now? If so, maybe it's better not to cherry-pick and make a minimal patch bumping the version only, so that the upstream changes can be picked up later.

File: contrib/depends/packages/ldns.mk (hunk 8 above)
Comment (stoffu) [inline contrib/depends/packages/ldns.mk:2]: This patch also looks different from the original, and this particular change even doesn't exist in Monero. If it's your original, please make a PR on Monero first.

If your cherry-picking results in a lot of merge conflicts and you have to modify the original patch significantly, then consider redoing it in the original order.

**Comment by stoffu** [comment]
Why do you need these updates? I'm OK with small patches that were cherry-picked without any merge conflicts, but you made some big patches which look like cherry-picked, but actually have some differences from the original.

Because dependency updates tend to make individual patches quite inter-dependent, I prefer merging them in the original order, unless the update is really mandatory (due to bugs etc).

**Comment by ghost** [comment]
It is not about me needing this update, I want to help with the Aeon project. No one is here to give me guidance on how to help so I am just doing whatever I feel is right. 

You have been mostly absent over the past many months, so I assumed you had moved on to other things. With no one working to merge upstream, I wanted to see if I can give help. So thanks for your returning, I will follow your review very carefully and hopefully it can be merged sometime.

File: contrib/depends/packages/unwind.mk (hunk 1 above)
Comment (ghost) [reply contrib/depends/packages/unwind.mk:2]: My bad, that is for https://github.com/libunwind/libunwind/pull/166 . Recommendation is to update to 1.5.0 . I will look into testing monero.

File: contrib/depends/packages/expat.mk (hunk 2 above)
Comment (ghost) [reply contrib/depends/packages/expat.mk:3]: Resolved.

File: contrib/depends/packages/qt.mk (hunk 3 above)
Comment (ghost) [reply contrib/depends/packages/qt.mk:2]: Resolved.

File: CMakeLists.txt (hunk 4 above)
Comment (ghost) [reply CMakeLists.txt:443]: Sorry not trying to be sneaky but I understand why the uneasiness. I will break that up to be more transparent.

File: contrib/depends/funcs.mk (hunk 6 above)
Comment (ghost) [reply contrib/depends/funcs.mk:134]: Resolved.

File: contrib/depends/packages/unwind.mk (hunk 9 above)
Comment (ghost) [reply contrib/depends/packages/unwind.mk:8]: Resolved.

File: contrib/depends/packages/openssl.mk (hunk 10 above)
Comment (ghost) [reply contrib/depends/packages/openssl.mk:3]: I see what you are saying now. We have no arm building support so we should not cherry pick this until we have merged that code. Then there will be easier picking in the future.  Well perhaps this is not a critical concern and can be postponed. I don't know how this library is used in the aeon code maybe it is a very insignificant dependency. hyc posted this comment about a potential flaw 1.0.2 https://github.com/monero-project/monero/pull/7100#issuecomment-744709797

File: contrib/depends/packages/hidapi.mk (hunk 7 above)
Comment (ghost) [reply contrib/depends/packages/hidapi.mk:5]: Resolved.

File: contrib/depends/packages/openssl.mk (hunk 10 above)
Comment (stoffu) [reply contrib/depends/packages/openssl.mk:3]: Currently Aeon uses OpenSSL only for calculating SHA256, so this vulnerability doesn't apply. But it will become relevant when we take the SSL support patches from upstream, so it may be advisable to bump it soonish.

**Comment by stoffu** [comment]
Thank you for your willingness to help, I appreciate your good intention.

Yes I have been absent for quite a while due to being busy with my day job and life events, but I'm not done with Aeon and I've been checking posts on IRC/Reddit/Bitcointalk regularly. I have been wanting to resume merging upstream patches which became long overdue. Your recent action motivated me to get to it again, so thank you.

This PR looks mostly OK apart from a few issues as commented above (in particular, OpenSSL update could be dropped and handled later).

**Comment by ghost** [comment]
Not interested in this anymore.
